### PR TITLE
Sensor Refactor

### PR DESCRIPTION
**Pull request by JBassett**
In this PR we now persist all sensor data to the local DB.  This allows us to keep the UI and Home Assistant better in sync.  This also allows sensors to return data asynchronously and allows for more reliable update with those sensors.

### CODE REVIEW

File: app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt [hunk 1]
```diff
@@ -15,6 +16,31 @@ class SensorsSettingsFragment : PreferenceFragmentCompat() {
...
+                            else
+                                it.summary = sensorEntity.state + " " + basicSensor.unitOfMeasurement
+                            // TODO: Add the icon from mdi:icon?
+                        } else {
+                            it.summary = "Disabled"
+                        }
+                    }
+                }
+            }
+            handler.postDelayed(this, 10000)
```
Comment (dshokouhi) [inline app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt:40]: Does this mean if we keep the settings screen open we can get updates every 10 seconds into HA?

Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt:40]: That is correct!  We can consider doing something similar to the iOS app add add this to the webview activity as well to get more consistent updates when the application is open.

Comment (dshokouhi) [reply app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt:40]: That would be amazing for wall mounted tablets!!!

File: app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt [hunk 2]
```diff
@@ -105,18 +110,12 @@ class PhoneStateSensorManager : SensorManager {
...
-            if (slotIndex == 1) {
-                return sim_2.toSensorRegistration(
-                    displayName,
-                    "mdi:sim",
-                    attrs
-                )
-            }
-            return sim_1.toSensorRegistration(
+            onSensorUpdated(context,
+                sim_1,
```
Comment (skynetua) [inline app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt:114]: There should be `basicSimSensor` i'm not mistaken

File: app/src/main/java/io/homeassistant/companion/android/database/AppDatabase.kt [hunk 3]
```diff
@@ -124,5 +128,32 @@ abstract class AppDatabase : RoomDatabase() {
...
+                        it.put("device_class", "")
+                    })
+                }
+                database.execSQL("DROP TABLE IF EXISTS `sensors`")
+                database.execSQL("CREATE TABLE IF NOT EXISTS `sensors` (`id` TEXT NOT NULL, `enabled` INTEGER NOT NULL, `registered` INTEGER NOT NULL, `state` TEXT NOT NULL, `state_type` TEXT NOT NULL, `type` TEXT NOT NULL, `icon` TEXT NOT NULL, `name` TEXT NOT NULL, `device_class` TEXT, `unit_of_measurement` TEXT, PRIMARY KEY(`id`))")
+                sensors.forEach {
+                    database.insert("sensors", OnConflictStrategy.REPLACE, it)
+                }
+
+                database.execSQL("CREATE TABLE IF NOT EXISTS `sensor_attributes` (`sensor_id` TEXT NOT NULL, `name` TEXT NOT NULL, `value` TEXT NOT NULL, PRIMARY KEY(`sensor_id`, `name`))")
```
Comment (klejejs) [inline app/src/main/java/io/homeassistant/companion/android/database/AppDatabase.kt:155]: Shouldn't the cursor be closed or does it close automatically?

Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/database/AppDatabase.kt:155]: Good catch, I thought I read somewhere you didn't need to close it but I'll close it anyway.

File: app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt (hunk 2 above)
Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt:114]: Good catch!

**Pull request merged**